src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -162,6 +161,63 @@ namespace AppInstaller::CLI::Workflow
         context.Add<Execution::Data::SearchResult>(context.Get<Execution::Data::Source>()->Search(searchRequest));
     }
 
+    void SearchSourceForSingle(Execution::Context& context)
+    {
+        const auto& args = context.Args;
+
+        MatchType matchType = MatchType::CaseInsensitive;
+        if (args.Contains(Execution::Args::Type::Exact))
+        {
+            matchType = MatchType::Exact;
+        }
+
+        SearchRequest searchRequest;
+        if (args.Contains(Execution::Args::Type::Query))
+        {
+            std::string_view query = args.GetArg(Execution::Args::Type::Query);
+            searchRequest.Inclusions.emplace_back(ApplicationMatchFilter(ApplicationMatchField::Id, matchType, query));
+            searchRequest.Inclusions.emplace_back(ApplicationMatchFilter(ApplicationMatchField::Name, matchType, query));
+            searchRequest.Inclusions.emplace_back(ApplicationMatchFilter(ApplicationMatchField::Moniker, matchType, query));
+        }
+
+        if (args.Contains(Execution::Args::Type::Id))
+        {
+            searchRequest.Filters.emplace_back(ApplicationMatchFilter(ApplicationMatchField::Id, matchType, args.GetArg(Execution::Args::Type::Id)));
+        }
+
+        if (args.Contains(Execution::Args::Type::Name))
+        {

```

Comment:
Shall we move the same logic to a common method? Or you expect them to divert in the future?